renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -260,6 +267,30 @@ static void create(WrappedVulkan *driver, const char *objName, const int line, V
     msaa.sampleShadingEnable = true;
   }
 
+  VkCompareOp stencilTest = VK_COMPARE_OP_ALWAYS;
+  VkStencilOp stencilOp = VK_STENCIL_OP_KEEP;
+  uint8_t stencilReference = 0;
+
+  switch(info.stencilOperations)
+  {
+    case StencilMode::KEEP:
+    {
+      break;
+    }
+    case StencilMode::KEEP_TEST_EQUAL_ONE:
+    {
+      stencilOp = VK_STENCIL_OP_KEEP;
+      stencilTest = VK_COMPARE_OP_EQUAL;
+      stencilReference = 1;
+      break;
+    }
+    case StencilMode::WRITE_ZERO:
+    {
+      stencilOp = VK_STENCIL_OP_ZERO;
```

Comment:
Looking at the changes here now I misspoke before, not every pipeline is explicitly writing zero as e.g. the `FillWithDiscardPattern` has a dynamic stencil ref, and so it's replacing with an arbitrary value (which it uses to fill the discard pattern in stencil)

I think it's better to keep the stencil op that was used here, `REPLACE`, rather than specialising to zero unnecessarily. It has the same effect but is more general.

Reply:
Thank you - good spot. I have kept `WRITE_ZERO` also even though it can be achieved with `REPLACE`. It felt more descriptive in its intent.